llvm/lib/Analysis/Lint.cpp

```diff
@@ -378,6 +382,36 @@ void Lint::visitReturnInst(ReturnInst &I) {
   }
 }
 
+bool Lint::isConstantAddressSpace(unsigned AS) const {
+  if (TT.isAMDGPU()) {
+    switch(AS) {
+      using namespace AMDGPUAS;
+    case CONSTANT_ADDRESS:
```

Comment:
Should this be a helper in AMDGPUAddrSpace.h? I'd expect this logic to already exist somewhere, but it looks like AMDGPUAA is only looking for CONSTANT_ADDRESS and nothing else: https://github.com/llvm/llvm-project/blob/97aa8cc94d94e8f0adc85489f7832ba7c0a9b577/llvm/lib/Target/AMDGPU/AMDGPUAliasAnalysis.cpp#L78